keel-web/src/test/kotlin/com/netflix/spinnaker/keel/apidocs/ApiDocTests.kt

```diff
@@ -80,232 +83,258 @@ class ApiDocTests : JUnit5Minutests {
+                    path("spec").isObject().path("\$ref").textValue().isEqualTo(constructRef(specSubType))
+                  }
+              }
+            }
         }
-      }
 
-    test("contains a schema for Constraint with all sub-types") {
-      at("/components/schemas/Constraint")
-        .isObject()
-        .path("oneOf")
-        .isArray()
-        .findValuesAsText("\$ref")
-        .containsExactlyInAnyOrder(
-          constructRef("CanaryConstraint"),
-          constructRef("DependsOnConstraint"),
-          constructRef("ManualJudgementConstraint"),
-          constructRef("PipelineConstraint"),
-          constructRef("TimeWindowConstraint")
-        )
-    }
+      resourceSpecTypes
+        .map(KClass<*>::simpleName)
+        .forEach { type ->
+          test("all properties of the parameterized version of the schema for Resource with a spec of $type are required") {
+            at("/components/schemas/${type}Resource/required")
+              .isArray()
+              .textValues()
+              .containsExactlyInAnyOrder("kind", "metadata", "spec")
+          }
+
+          test("the metadata property of the parameterized version of the schema for SubmittedResource with a spec of $type are required") {
+            at("/components/schemas/${type}SubmittedResource/required")
+              .isArray()
+              .textValues()
+              .containsExactlyInAnyOrder("kind", "spec")
+          }
 
-    sequenceOf(
-      CanaryConstraint::class,
-      DependsOnConstraint::class,
-      ManualJudgementConstraint::class,
-      PipelineConstraint::class,
-      TimeWindowConstraint::class
-    )
-      .map(KClass<*>::simpleName)
-      .forEach { type ->
-        test("Constraint sub-type $type has its own schema") {
-          at("/components/schemas/$type")
-            .isObject()
+          test("ResourceSpec sub-type $type has its own schema") {
+            at("/components/schemas/$type")
+              .isObject()
+          }
         }
+
+      test("contains a schema for Constraint with all sub-types") {
+        at("/components/schemas/Constraint")
+          .isObject()
+          .path("oneOf")
+          .isArray()
+          .findValuesAsText("\$ref")
+          .containsExactlyInAnyOrder(
+            constructRef("CanaryConstraint"),
+            constructRef("DependsOnConstraint"),
+            constructRef("ManualJudgementConstraint"),
+            constructRef("PipelineConstraint"),
+            constructRef("TimeWindowConstraint")
+          )
       }
 
-    test("contains a schema for DeliveryArtifact with all sub-types") {
-      at("/components/schemas/DeliveryArtifact")
-        .isObject()
-        .path("oneOf")
-        .isArray()
-        .findValuesAsText("\$ref")
-        .containsExactlyInAnyOrder(
-          constructRef("DebianArtifact"),
-          constructRef("DockerArtifact")
-        )
-    }
+      sequenceOf(
```

Comment:
It'd be nice to build these lists automatically to avoid missing newly introduced types in the future.